spec-tests/src/test/java/org/candlepin/spec/entitlements/StorageBandSpecTest.java

```diff
@@ -0,0 +1,256 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package org.candlepin.spec.entitlements;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import org.candlepin.dto.api.client.v1.AttributeDTO;
+import org.candlepin.dto.api.client.v1.ComplianceStatusDTO;
+import org.candlepin.dto.api.client.v1.ConsumerDTO;
+import org.candlepin.dto.api.client.v1.ConsumerInstalledProductDTO;
+import org.candlepin.dto.api.client.v1.EntitlementDTO;
+import org.candlepin.dto.api.client.v1.OwnerDTO;
+import org.candlepin.dto.api.client.v1.PoolDTO;
+import org.candlepin.dto.api.client.v1.ProductDTO;
+import org.candlepin.resource.client.v1.OwnerProductApi;
+import org.candlepin.spec.bootstrap.client.ApiClient;
+import org.candlepin.spec.bootstrap.client.ApiClients;
+import org.candlepin.spec.bootstrap.client.SpecTest;
+import org.candlepin.spec.bootstrap.client.api.OwnerClient;
+import org.candlepin.spec.bootstrap.data.builder.ConsumerTypes;
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.builder.Pools;
+import org.candlepin.spec.bootstrap.data.builder.Products;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+@SpecTest
+public class StorageBandSpecTest {
+
+    private static ApiClient client;
+    private static OwnerClient ownerApi;
+    private static OwnerProductApi ownerProductApi;
+
+    @BeforeAll
+    public static void beforeAll() {
+        client = ApiClients.admin();
+        ownerApi = client.owners();
+        ownerProductApi = client.ownerProducts();
+    }
+
+    @Test
+    public void shouldPoolHaveTheCorrectQuantityBasedOffOfTheProductMultiplier() {
+        // sub.quantity * multiplier
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ApiClient userClient = ApiClients.basic(UserUtil.createUser(client, owner));
+        ProductDTO cephProduct = createCephProductAndPool(owner);
+
+        assertEquals(512, userClient.pools().listPoolsByOwnerAndProduct(owner.getId(),
+            cephProduct.getId()).get(0).getQuantity());
```

Comment:
We should assert that the size of the list returned from listPoolsByOwnerAndProduct is 1 before doing a `get(0)` to avoid IndexOutOfBoundsException.